infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -489,6 +493,48 @@ public synchronized void saveManagementLayout(Layout layout) {
         }
     }
 
+    public synchronized void saveFailureDetectorMetrics(FailureDetectorMetrics detector) {
+        boolean enabled = Boolean.parseBoolean(System.getProperty("corfu.failuredetector", Boolean.FALSE.toString()));
+        if (!enabled){
+            return;
+        }
+
+        dataStore.put(
+                FailureDetectorMetrics.class,
+                PREFIX_FAILURE_DETECTOR,
+                String.valueOf(getManagementLayout().getEpoch()),
+                detector
+        );
+    }
+
+    public FailureDetectorMetrics getFailureDetectorMetrics() {
```

Comment:
Comments!
Also maybe talk to team about how to store historical data.

Reply:
ok